euphonic_wrapper.py

```diff
@@ -0,0 +1,277 @@
+import numpy as np
+import os
+from euphonic import Crystal, ForceConstants, QpointPhononModes, DebyeWaller, ureg
+from euphonic.util import mp_grid
+
+class euphonic_wrapper(object):
+    # This a wrapper around the Euphonic ForceConstants and QpointPhononModes classes to make it easier to access from Matlab
+    # It is meant to be used with a Matlab python_wrapper class and implements a horace_sqw function for use with Horace
+
+    def __init__(self, value=None, **kwargs):
+        self.force_constants = None
+        self.phonon_modes = None
+        self.debye_waller = None
+        self.debye_waller_grid = None
+        self.temperature = 0.0 * ureg('K')
+        self.bose = True
+        self.negative_e = False
+        self.conversion_mat = None
+        self.chunk = 5000
+        self.lim = np.inf
+        self.scattering_lengths = 'Sears1992'
+        self.weights = None
+        self.asr = None
+        self.dipole = True
+        self.eta_scale = 1.0
+        self.splitting = True
+        self.insert_gamma = False
+        self.reduce_qpts = True
+        self.use_c = False
+        self.n_threads = 1
+        self.fall_back_on_python = True
+        
+        if value is not None:
+            if isinstance(value, str):
+                self.load(value)
+            elif hasattr('calculate_qpoint_phonon_modes'):
+                self.force_constants = value
+            elif hasattr('calculate_structure_factor'):
+                self.phonon_modes = value
```

Comment:
Do you mean `hasattr(value, 'function')`? And is there a use case for users inputting a `QpointPhononModes` file directly? Is this something we want/need to support?

Reply:
Oh yeah... Doh!